db/migrate/20231011103114_migrate_notification_to_noticed.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class MigrateNotificationToNoticed < ActiveRecord::Migration[7.0]
+  def up
+    add_column :notifications, :params, :jsonb, default: {}, null: false
+    add_column :notifications, :type, :string
+    add_column :notifications, :read_at, :datetime
+    add_column :notifications, :recipient_id, :bigint
+    add_column :notifications, :recipient_type, :string
```

Comment:
use `add_column :recipient, :references, polymorphic: true`, so it will automatically create indexes